jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6132 @@
+	 * @param setValue value that will be set in obj at offset if compare is successful
+	 * @return boolean value indicating whether the field was updated
+	 */
+	public final boolean weakCompareAndSetIntAcquire(Object obj, long offset, int compareValue, int setValue) {
+		return compareAndSetInt(obj, offset, compareValue, setValue);
+	}
+
+	/**
+	 * Sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of setRelease.
+	 * 
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param setValue value that will be set in obj at offset if compare is successful
+	 * @return boolean value indicating whether the field was updated
+	 */
+	public final boolean weakCompareAndSetIntRelease(Object obj, long offset, int compareValue, int setValue) {
+		return compareAndSetInt(obj, offset, compareValue, setValue);
+	}
+
+	/**
+	 * Sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of set.
+	 * 
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param setValue value that will be set in obj at offset if compare is successful
+	 * @return boolean value indicating whether the field was updated
+	 */
+	public final boolean weakCompareAndSetInt(Object obj, long offset, int compareValue, int setValue) {
+		return compareAndSetInt(obj, offset, compareValue, setValue);
+	}
+
+	/**
+	 * Atomically sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of set.
+	 *
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param exchangeValue value that will be set in obj at offset if compare is successful
+	 * @return value of the field before this operation. This will be compareValue if the exchange was successful
+	 */
+	public final byte compareAndExchangeByte(Object obj, long offset, byte compareValue, byte exchangeValue) {
+		return compareAndExchange8bits(obj, offset, compareValue, exchangeValue);
+	}
+
+	/**
+	 * Atomically sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of set.
+	 * 
```

Comment:
This has the same doc as the weak variant, should they be different?

Reply:
These should have getVolatile and setVolatile for memory semantics instead of get and set.